orm/db_alias.go

```diff
@@ -402,3 +458,101 @@ func GetDB(aliasNames ...string) (*sql.DB, error) {
 	}
 	return nil, fmt.Errorf("DataBase of alias name `%s` not found", name)
 }
+
+type stmtUtil struct {
+	refCount int
+	//only from false to true
+	isClosed bool
+	sqlStmt *sql.Stmt
+	signClose chan bool
+
+	sync.Mutex
+}
+
+func (s *stmtUtil) getStmt() (*sql.Stmt,error) {
+	s.Lock()
+	defer s.Unlock()
+
+	if s.isClosed {
+		return nil, errors.New("stmtUtil closed")
+	}
+
+	return s.sqlStmt, nil
+}
+
+func (s *stmtUtil) occupy() error {
+	s.Lock()
+	defer s.Unlock()
+
+	if s.isClosed {
+		return errors.New("stmtUtil closed")
+	}
+
+	s.refCount++
+	return nil
+}
+
+func (s *stmtUtil) release()  {
+	s.Lock()
+
+	if s.isClosed {
+		s.Unlock()
+		return
+	}
+
+	closedSign := false
+	if s.refCount >= 1 {
+		s.refCount --
+	}
+	if s.refCount <= 0 {
+		closedSign = true
+	}
+	s.Unlock()
+
+	go func() {
+		s.signClose <- closedSign
+	}()
+}
+
+//garbage recycle for stmt
+func (s *stmtUtil) gc() {
+	for {
+		select {
+		case closed := <-s.signClose:
+			if !closed {
+				continue
+			}
+			time.AfterFunc(60*time.Second, func() {
```

Comment:
As my understanding, if there is a invocation sequence like (occupy, release, occupy, release...), we will register many AfterFunc callbacks. But most of them will not work since the refCount > 0;
Actually, in this callback, we don't need to check anything, close stmt directly because this is the final backup way to close stmt.